### PR TITLE
[fix][broker] do not serialize field bundleStats

### LINKED ISSUE #17738: Duplicate data stored in loadbalance/brokers/
### Search before asking

- [x] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Motivation

i find that method getBundleStats return member lastStats, causing  duplicate data stored in ZK when Jackson serializing LocalBrokerData, and the size of data in lastStats is pretty large, consisting of metrics data concerning to every bundles that belong to this broker, and i find that the method LocalBrokerData.getBundleStats is not used anywhere, i wonder whether we can deduplicate that. 

org.apache.pulsar.policies.data.loadbalancer.LocalBrokerData#getBundleStats
image

we could verify this by retrieving data from ZK.
image
image

### Solution

do not serialize field bundleStats.

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by thetumbled**
Fixes #17738

### Modifications
because broker will do not use the field bundleStats, so we could do not serialize the field bundleStats.

### Verifying this change
This change is a trivial rework / code cleanup without any test coverage.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: 
https://github.com/thetumbled/pulsar/pull/3

### CODE REVIEW

**Review by codelipenghui — COMMENTED** [review]
Nice catch!
Could you please help add an unit test to make sure we will not break it again in the future?

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+@Test(groups = "broker")
+public class LocalBrokerDataTest {
+    /*
+    Ensuming that there no bundleStats field in the json string serialized from LocalBrokerData.
+     */
+    @Test
+    public void testSerializeLocalBrokerData() throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        LocalBrokerData localBrokerData = new LocalBrokerData();
+        assert !objectMapper.writeValueAsString(localBrokerData).contains("bundleStats");
```
Comment (Demogorgon314) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java:19]: Use `org.testng.Assert.assertTrue` here. 

Or
```
assertFalse(objectMapper.writeValueAsString(localBrokerData).contains("bundleStats"));
```

**Comment by thetumbled** [comment]
> Nice catch! Could you please help add an unit test to make sure we will not break it again in the future?

done.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java (hunk 1 above)
Comment (thetumbled) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java]: done.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package org.apache.pulsar.broker.loadbalance;
```
Comment (Demogorgon314) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java:1]: The new test class needs to add an apache license header. You can copy it from other classes.

Comment (thetumbled) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java:19]: done.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.policies.data.loadbalancer.LocalBrokerData;
+import org.testng.annotations.Test;
+
+@Slf4j
+@Test(groups = "broker")
+public class LocalBrokerDataTest {
+    /*
+    Ensuming that there no bundleStats field in the json string serialized from LocalBrokerData.
+     */
```
Comment (Demogorgon314) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java:33]: nit: `Ensuming` -> `Ensuring`
```suggestion
    /**
     * Ensuring that there no bundleStats field in the json string serialized from LocalBrokerData.
     */
```

Comment (thetumbled) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java]: done.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.policies.data.loadbalancer.LocalBrokerData;
+import org.testng.annotations.Test;
+
+@Slf4j
+@Test(groups = "broker")
+public class LocalBrokerDataTest {
+    /*
+    Ensuring that there no bundleStats field in the json string serialized from LocalBrokerData.
+     */
```
Comment (RobertIndie) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java:33]:
```suggestion
    /**
     * Ensure that there is no bundleStats field in the json string serialized from LocalBrokerData.
     */
```

Comment (thetumbled) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LocalBrokerDataTest.java]: i found another LocalBrokerDataTest class, so i move the implementation into it.

**Review by mattisonchao — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Review by nodece — APPROVED** [review]

**Review by Demogorgon314 — APPROVED** [review]

**Pull request merged**